src/components/dashboard/main-content/course-enrollments/tests/CourseEnrollments.test.jsx

```diff
@@ -60,28 +58,39 @@ function renderEnrollmentsComponent(initialProps) {
     </Provider>,
   );
 }
+describe('Course enrollements', () => {
+  beforeEach(() => {
+    updateCourseCompleteStatusRequest.mockImplementation(() => {});
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  test('loads enrollments component', () => {
+    renderEnrollmentsComponent(initProps);
+    expect(screen.getByText('My courses in progress')).toBeInTheDocument();
+    expect(screen.getByText('Completed courses')).toBeInTheDocument();
+    expect(screen.getByText('Courses saved for later')).toBeInTheDocument();
+    expect(screen.getAllByText('edX Demonstration Course').length).toBeGreaterThanOrEqual(1);
+  });
 
-test('loads enrollments component', () => {
-  renderEnrollmentsComponent(initProps);
-  expect(screen.getByText('My courses in progress')).toBeInTheDocument();
-  expect(screen.getByText('Courses saved for later')).toBeInTheDocument();
-  expect(screen.getAllByText('edX Demonstration Course').length).toBeGreaterThanOrEqual(1);
-});
+  it('generates course status update on move to in progress action', () => {
+    renderEnrollmentsComponent({ ...initProps });
+    expect(screen.getByRole('button', { name: 'Move course to In Progress' })).toBeInTheDocument();
+    fireEvent.click(screen.getByRole('button', { name: 'Move course to In Progress' }));
+
+    // TODO This test only validates 'half way', we ideally want to update it to
+    // validate the UI results. Skipping at the time of writing since need to
+    // figure out the right markup for testability. This give a base level of confidence
+    // that move to in progress is not failing, that's all.
+    expect(updateCourseCompleteStatusRequest).toHaveBeenCalledTimes(1);
+  });
+
+  it('generates course status update on move to saved for later action', () => {
+    renderEnrollmentsComponent({ ...initProps });
```

Comment:
ditto: let's go with just initProps here? the same results I think, but just  think that spreading here is not adding anything

Reply:
So, the reason I tend to go for `{ ...initProps}` is because it guarantees that whatever I'm passing it to will not change the initial object. And/or use something like the createCompletedCourse function that we use elsewhere, to ensure we've got a new, unmodified object every time. I've been burned too many times by objects that were modified and therefore not what I thought they were.

Reply:
I see yeah now that I think of it yes that is better indeed, thanks for this clarification. yes let's always pass in the copy (it's basically a shallow copy but yes that is always cleaner), so yeah let's use this pattern in the entire test